src/NzbDrone.Common/Http/HttpUri.cs

```diff
@@ -144,23 +144,38 @@ private static string CombinePath(string basePath, string relativePath)
                 return basePath;
             }
 
-            var newPath = "/" + relativePath.TrimStart('/');
+            if (basePath.IsNullOrWhiteSpace())
+            {
+                return relativePath;
+            }
+
+            return basePath.TrimEnd('/') + "/" + relativePath.TrimStart('/');
+        }
 
-            if (basePath != null && !relativePath.StartsWith("/"))
+        private static string CombineRelativePath(string basePath, string relativePath)
+        {
+            if (relativePath.IsNullOrWhiteSpace())
             {
-                var baseSlashIndex = basePath.LastIndexOf('/');
+                return basePath;
+            }
 
-                if (baseSlashIndex == basePath.Length - 1)
-                {
-                    newPath = basePath.TrimEnd('/') + newPath;
-                }
-                else if (baseSlashIndex != 0)
-                {
-                    newPath = basePath.Substring(0, baseSlashIndex) + newPath;
-                }
+            if (relativePath.StartsWith("/"))
+            {
+                return relativePath;
             }
 
-            return newPath;
+            var baseSlashIndex = basePath.LastIndexOf('/');
+
+            if (baseSlashIndex == 0)
```

Comment:
No, if there's a slash in base then everything after it gets removed. (the last slash).
During some coding substring(0,0) seems no good, but that's not true. I'll clean it up a bit to, but functionally that's still the same:
```
if (baseSlashIndex >= 0)
{
   return basePath.Substring(0, baseSlashIndex) + "/" + relativePath.TrimStart('/');
}
```